include/reactor-cpp/port.hh

```diff
@@ -18,122 +19,112 @@ namespace reactor {
   friend class Scheduler;
 };
 
-template <class T>
-class Port : public BasePort {
- private:
-  ImmutableValuePtr<T> value_ptr{nullptr};
+template <class T> class Port : public BasePort {
+private:
+  ImmutableValuePtr<T> value_ptr_{nullptr};
 
-  void cleanup() override final { value_ptr = nullptr; }
+  void cleanup() final { value_ptr_ = nullptr; }
 
- public:
+public:
   using value_type = T;
 
   Port(const std::string& name, PortType type, Reactor* container)
       : BasePort(name, type, container) {}
 
   void bind_to(Port<T>* port) { base_bind_to(port); }
-  Port<T>* typed_inward_binding() const;
-  const std::set<Port<T>*>& typed_outward_bindings() const;
+  auto typed_inward_binding() const noexcept -> Port<T>*;
+  auto typed_outward_bindings() const noexcept -> const std::set<Port<T>*>&;
 
   void set(const ImmutableValuePtr<T>& value_ptr);
-  void set(MutableValuePtr<T>&& value_ptr) {
-    set(ImmutableValuePtr<T>(std::forward<MutableValuePtr<T>>(value_ptr)));
-  }
+  void set(MutableValuePtr<T>&& value_ptr) { set(ImmutableValuePtr<T>(std::forward<MutableValuePtr<T>>(value_ptr))); }
   void set(const T& value) { set(make_immutable_value<T>(value)); }
   void set(T&& value) { set(make_immutable_value<T>(std::forward<T>(value))); }
   // Setting a port to nullptr is not permitted.
   void set(std::nullptr_t) = delete;
 
-  void startup() override final {}
-  void shutdown() override final {}
-
-  const ImmutableValuePtr<T>& get() const;
-  bool is_present() const;
+  void startup() final {}
+  void shutdown() final {}
 
+  auto get() const noexcept -> const ImmutableValuePtr<T>&;
+  [[nodiscard]] auto is_present() const noexcept -> bool;
 };
 
-template <>
-class Port<void> : public BasePort {
- private:
-  bool present{false};
-
-  void cleanup() override final { present = false; }
+template <> class Port<void> : public BasePort {
+private:
+  bool present_ = false;
 
- public:
-  using value_type = void;
+  void cleanup() final { present_ = false; }
 
+public:
   Port(const std::string& name, PortType type, Reactor* container)
       : BasePort(name, type, container) {}
 
   void bind_to(Port<void>* port) { base_bind_to(port); }
-  Port<void>* typed_inward_binding() const;
-  const std::set<Port<void>*>& typed_outward_bindings() const;
+  [[nodiscard]] auto typed_inward_binding() const noexcept -> Port<void>*;
+  [[nodiscard]] auto typed_outward_bindings() const noexcept -> const auto&;
 
   void set();
-  bool is_present() const;
+  [[nodiscard]] auto is_present() const noexcept -> bool;
 
-  void startup() override final {}
-  void shutdown() override final {}
+  void startup() final {}
+  void shutdown() final {}
 };
 
-template <class T>
-class Input : public Port<T> {
- public:
+template <class T> class Input : public Port<T> { // NOLINT
+public:
   Input(const std::string& name, Reactor* container)
       : Port<T>(name, PortType::Input, container) {}
-
-  Input(Input&&) = default;
+  Input(Input&&) noexcept = default;
```

Comment:
Could it be that this change triggers the build problems with gcc 9.3? For some reason, the compiler rejects the Input type as not move-assignable. This change here modifies the constructor signature, and it might be the reason why the constructor is not detected as a move-constructor.